src/sentry/api/endpoints/project_rule_preview.py

```diff
@@ -0,0 +1,68 @@
+from ast import literal_eval
+
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectAlertRulePermission, ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.api.serializers.rest_framework.rule import RuleSetSerializer
+from sentry.rules.history.endpoints.project_rule_stats import TimeSeriesValueSerializer
+from sentry.rules.history.preview import preview
+from sentry.web.decorators import transaction_start
+
+
+@region_silo_endpoint
+class ProjectRulePreviewEndpoint(ProjectEndpoint):
+    private = True
+    permission_classes = (ProjectAlertRulePermission,)
+
+    @transaction_start("ProjectRulePreviewEndpoint")
+    def get(self, request: Request, project) -> Response:
+        """
+        Get a list of alert triggers in past 2 weeks for given rules
+
+            {method} {path}
+            {{
+                "conditions": [],
+                "filters": [],
+                "actionMatch": "all",
+                "filterMatch": "all",
+                "frequency": 60,
+            }}
+
+        """
+        data = {
```

Comment:
I don't think you need toe extract these arguments from the query parameters. you should be able to just pass in `request.GET` into `RuleSetSerializer` and the validated_data should contain everything you need.

Reply:
It doesn't seem to be able to parse a `RuleNodeField` from a string, also can't find any examples of other get endpoints that need to parse a list of maps

Reply:
You might need to make a new custom field that does the parsing. But totally understand it being outside the scope of this PR. Put a TODO down, make a jira ticket (or something) and I'll mark this as resolved.

Reply:
I think that should do it, I hope `literal_eval` is safe, but let me know if we have our own method to do something like this

Reply:
LGTM!